nemo/collections/nlp/models/token_classification/ner_model.py

```diff
@@ -0,0 +1,215 @@
+
+        self.bert_model = get_pretrained_lm_model(
+            pretrained_model_name=cfg.language_model.pretrained_model_name,
+            config_file=cfg.language_model.bert_config,
+            checkpoint_file=cfg.language_model.bert_checkpoint,
+        )
+        self.hidden_size = self.bert_model.config.hidden_size
+
+        self.classifier = TokenClassifier(
+            hidden_size=self.hidden_size,
+            num_classes=self.data_desc.num_classes,
+            num_layers=cfg.head.num_fc_layers,
+            activation=cfg.head.activation,
+            log_softmax=cfg.head.log_softmax,
+            dropout=cfg.head.fc_dropout,
+            use_transformer_init=cfg.head.use_transformer_init,
+        )
+
+        if cfg.class_balancing == 'weighted_loss':
+            # You may need to increase the number of epochs for convergence when using weighted_loss
+            self.loss = CrossEntropyLoss(weight=self.data_desc.class_weights)
+        else:
+            self.loss = CrossEntropyLoss()
+
+        # setup to track metrics
+        self.class_report = ClassificationReport(self.data_desc.num_classes, label_ids=self.data_desc.label_ids)
+        # Optimizer setup needs to happen after all model weights are ready
+        self.setup_optimization(cfg.optim)
+
+    @typecheck()
+    def forward(self, input_ids, token_type_ids, attention_mask):
+        hidden_states = self.bert_model(
+            input_ids=input_ids, token_type_ids=token_type_ids, attention_mask=attention_mask
+        )
+        logits = self.classifier(hidden_states=hidden_states)
+        return logits
+
+    def training_step(self, batch, batch_idx):
+        """
+        Lightning calls this inside the training loop with the data from the training dataloader
+        passed in as `batch`.
+        """
+        # forward pass
+        input_ids, input_type_ids, input_mask, loss_mask, subtokens_mask, labels = batch
+        logits = self(input_ids=input_ids, token_type_ids=input_type_ids, attention_mask=input_mask)
+
+        loss = self.loss(logits=logits, labels=labels, loss_mask=loss_mask)
+        tensorboard_logs = {'train_loss': loss, 'lr': self._optimizer.param_groups[0]['lr']}
+        return {'loss': loss, 'log': tensorboard_logs}
+
+    def validation_step(self, batch, batch_idx):
+        """
+        Lightning calls this inside the validation loop with the data from the validation dataloader
+        passed in as `batch`.
+        """
+        input_ids, input_type_ids, input_mask, loss_mask, subtokens_mask, labels = batch
+        logits = self(input_ids=input_ids, token_type_ids=input_type_ids, attention_mask=input_mask)
+        val_loss = self.loss(logits=logits, labels=labels, loss_mask=loss_mask)
+
+        subtokens_mask = subtokens_mask > 0.5
+
+        preds = torch.argmax(logits, axis=-1)[subtokens_mask]
+        labels = labels[subtokens_mask]
+        tp, fp, fn = self.class_report(preds, labels)
```

Comment:
How about classification_report instaed of class_report?

Reply:
done